File: app/src/main/java/com/cajetan/youtubeplus/data/VideoDatabase.java
package com.cajetan.youtubeplus.data;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

@Database(entities = {VideoData.class}, version = 1, exportSchema = false)
public abstract class VideoDatabase extends RoomDatabase {
    public abstract VideoDataDao videoDataDao();

    private static volatile VideoDatabase INSTANCE;

    static VideoDatabase getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (VideoDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            VideoDatabase.class, "video_database").build();
                }
            }
        }

        return INSTANCE;
    }
}

File: app/src/main/java/com/cajetan/youtubeplus/data/VideoDatabase.kt
package com.cajetan.youtubeplus.data

import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import android.content.Context

@Database(entities = [VideoData::class, PlaylistData::class], version = 2, exportSchema = false)
abstract class VideoDatabase : RoomDatabase() {
    abstract fun videoDataDao(): VideoDataDao
    abstract fun playlistDataDao(): PlaylistDataDao

    companion object {
        private var INSTANCE: VideoDatabase? = null

        fun getDatabase(context: Context): VideoDatabase? {
            if (INSTANCE == null) {
                synchronized(VideoDatabase::class) {
                    if (INSTANCE == null) {
                        INSTANCE = Room.databaseBuilder(context.applicationContext,
                                VideoDatabase::class.java, "video_database")
                                .fallbackToDestructiveMigration().build()
                    }
                }
            }

            return INSTANCE
        }
    }
}

